src/arithmetic/arithmetic_expression.c

```diff
@@ -978,6 +1004,35 @@ SIValue AR_TIMESTAMP(SIValue *argv, int argc) {
 	return SI_LongVal(TemporalValue_NewTimestamp());
 }
 
+/* Conditional flow functions */
+/* Case When
+ * Case Value [When Option i Then Result i] Else Default end */
+SIValue AR_CASEWHEN(SIValue *argv, int argc) {
+	/* Expecting Value and Default.
+	 * Assuming values maintain original specified order. */
+	assert(argc > 1);
+
+	/* argv[0] - Value
+	 * argv[i] - Option i
+	 * argv[i+1] - Result i
+	 * argv[argc-1] - Default */
+	SIValue v = argv[0];
+	SIValue d = argv[argc - 1];
+
+	int alternatives = argc - 1;
+	for(int i = 1; i < alternatives; i += 2) {
+		SIValue a = argv[i];
+		if(SIValue_Compare(v, a) == 0) {
+			// Return Result i.
+			return argv[i + 1];
+		}
+	}
+
+	/* Did not match against any Option
+	 * Return default. */
+	return d;
```

Comment:
Is d defaults to null?

Reply:
yep, if default is not specified it is defaults to NULL